components/core/src/package/target.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::result;
+use std::str::FromStr;
+use os::system::{uname, Uname};
+
+use error::{Error, Result};
+
+pub trait Target: fmt::Display + Into<PackageTarget> {
+    fn platform(&self) -> &str;
+    fn architecture(&self) -> &str;
+    fn valid_for(&self, uname: Uname) -> Result<()>;
+}
+
+/// Describes the platform (operating system/kernel)
+/// and architecture (x86_64, i386, etc..) that a package is built for
+#[derive(RustcEncodable, RustcDecodable, Debug, Clone, Hash)]
+pub struct PackageTarget {
+    pub platform: String,
```

Comment:
I think it's a bit more appropriate for both the platform and architecture should be represented as enums instead of Strings. It's a bit more performant but more importantly, anytime we can use the type system we should ;)